src/ResourceManager/RedisCache/Commands.RedisCache/Commands/NewAzureRedisCacheFirewallRule.cs

```diff
@@ -0,0 +1,113 @@
+    using System.Management.Automation;
+    using Rest.Azure;
+
+    [Cmdlet(VerbsCommon.New, "AzureRmRedisCacheFirewallRule", DefaultParameterSetName = NormalParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSRedisFirewallRule))]
+    public class NewAzureRedisCacheFirewallRule : RedisCacheCmdletBase
+    {
+        private const string NormalParameterSet = "NormalParameterSet";
+        private const string InputObjectParameterSet = "RedisCacheAttributesObject";
+        private const string ResourceIdParameterSet = "ResourceIdParameterSet";
+
+        [Parameter(ParameterSetName = NormalParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = false, HelpMessage = "Name of resource group in which cache exists.")]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = NormalParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Name of redis cache.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = NormalParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Name of firewall rule.")]
+        [Parameter(ParameterSetName = InputObjectParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Name of firewall rule.")]
+        [Parameter(ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Name of firewall rule.")]
+        [ValidateNotNullOrEmpty]
+        public string RuleName { get; set; }
+
+        [Parameter(ParameterSetName = NormalParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Starting IP address.")]
+        [Parameter(ParameterSetName = InputObjectParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Starting IP address.")]
+        [Parameter(ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Starting IP address.")]
+        [ValidateNotNullOrEmpty]
+        public string StartIP { get; set; }
+
+        [Parameter(ParameterSetName = NormalParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Ending IP address.")]
+        [Parameter(ParameterSetName = InputObjectParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Ending IP address.")]
+        [Parameter(ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Mandatory = true, HelpMessage = "Ending IP address.")]
+        [ValidateNotNullOrEmpty]
+        public string EndIP { get; set; }
+
+        [Parameter(ParameterSetName = InputObjectParameterSet, Mandatory = true, ValueFromPipeline = true, HelpMessage = "object of type RedisCacheAttributes")]
+        [ValidateNotNull]
+        public RedisCacheAttributes InputObject { get; set; }
```

Comment:
Please move InputObject and ResourceId above Rule Name in the order - this will make the parameter set in the markdown file more logical.